### PR TITLE
Embed Modality into a separate module

### PR DESCRIPTION
**Pull request by enver1323**
This PR embeds modality into a separate Module.
This allows to separate all modality-conditional statement into different classes.

**TODO**:
- [x] Move feature-mask selection into modality
- [x] Check postprocessors code

### CODE REVIEW

**Review by ghkim9213 — APPROVED** [review]
What a improvement!

File: pnpxai/evaluator/optimizer/search_spaces.py
[the remark this answers is not in the document]
Comment (enver1323) [reply pnpxai/evaluator/optimizer/search_spaces.py:1]: Thank you for your suggestion. I removed this file in the latest commit

File: pnpxai/core/recommender/recommender.py [hunk 1]
```diff
@@ -148,86 +63,22 @@ def print_tabular(self):
...
-                map_data[modality].add(explainer_type)
-        self.modality_to_explainers_map = RecommendationMap(map_data, ["modality", "explainers"])
+        self.architecture_to_explainers_map = self._build_architecture_to_explainers_map()
 
     def _build_architecture_to_explainers_map(self):
         map_data = {}
-        for explainer_type, v in self._map_data.items():
-            for arch in v["module_types"]:
+        for explainer_type in AVAILABLE_EXPLAINERS:
+            for arch in explainer_type.SUPPORTED_MODULES:
```
[the remark this answers is not in the document]
Comment (enver1323) [reply pnpxai/core/recommender/recommender.py:71]: Thank you!

File: pnpxai/evaluator/optimizer/suggestor.py
Comment (ghkim9213) [inline pnpxai/evaluator/optimizer/suggestor.py:1]: 👍

**Comment by ghkim9213** [comment]
@enver1323 #128 Here are some modifications for text and vqa modality.

**Comment by enver1323** [comment]
Thank you, very much for your help!

**Pull request merged**